src/Analysis/Ast/Impl/Types/PythonTypeWrapper.cs

```diff
@@ -32,7 +34,11 @@ protected IPythonType InnerType
         /// Creates delegate type wrapper over an existing type.
         /// Use dedicated constructor for wrapping builtin types.
         /// </summary>
-        public PythonTypeWrapper(IPythonType type) : this(type, type.DeclaringModule) {
+        public PythonTypeWrapper(IPythonType type) : this(type, type.DeclaringModule) { }
+
+        public PythonTypeWrapper(string typeName, string documentation, IPythonModule declaringModule, IPythonType baseType) : this(baseType, declaringModule) {
```

Comment:
Allow override name and documentation in the wrapper. Comes handy in some type restoration cases, such as with named tuples.